suprb/optimizer/rule/generation_operator/mutation.py

```diff
@@ -1,35 +1,24 @@
-from abc import ABCMeta, abstractmethod
 
 from typing import Union
 
 import numpy as np
 from scipy.stats import halfnorm
 
-from suprb.base import BaseComponent
 from suprb.rule import Rule
 from suprb.utils import RandomState
-from suprb.rule.matching import MatchingFunction, OrderedBound
+from suprb.optimizer.rule.generation_operator import GenerationOperator
+from suprb.rule.matching import MatchingFunction
 
 
-class RuleMutation(BaseComponent, metaclass=ABCMeta):
+class RuleMutation(GenerationOperator):
```

Comment:
Isn't this still an ABC? Or may we not do that as we inherit from an ABCMeta?

Reply:
That's how I understand inheritance. It should be the case here as well

Reply:
standard way is to keep it as is but make as many of the methods abstract as possible